core/domain/question_services.py

```diff
@@ -142,21 +142,26 @@ def get_questions_and_skill_descriptions_by_skill_ids(
     return questions, skill_descriptions, next_cursor
 
 
-def get_questions_by_skill_ids(question_count, skill_ids):
-    """Returns constant number of questions linked to the given skill ids.
+def get_questions_by_skill_ids(total_question_count, skill_ids):
+    """Returns constant number of questions linked to each given skill ids.
 
     Args:
-        question_count: int. The number of questions to return.
-        skill_ids: list(str). The ID of the skills to which the questions are
-            linked.
+        total_question_count: int. The total number of questions to return.
+        skill_ids: list(str). The IDs of the skills to which the questions
+            should be linked.
 
     Returns:
-        list(Question). The list of constant number of questions linked to each
-            given skill id.
+        list(Question). The list containing an expected number of
+        total_question_count questions linked to each given skill id.
+        It will return more if total_question_count is not evenly divisable,
+        or less if not enough questions in datastore.
     """
+    total_question_count = min(
+        total_question_count, feconf.MAX_QUESTIONS_FETCHABLE_AT_ONE_TIME)
```

Comment:
No, do a hard reject if you are asked for too many questions.

Actually, I realize I misspoke in my previous review, sorry. Do this check at the controller layer instead, and raise a 400 (invalid input) error instead if the number of questions asked for is too large. That way we reject it at the server API boundary and prevent that issue creeping further into the system.

Then, at this level, do a similar check but use an if statement and raise Exception if the requested count is above the limit, since nothing should ever get through to this point in normal system operation.

Reply:
Again, I think this is out of scope of this PR because this should be a question player config change rather than a review test specific one no?

Reply:
No, it's specific to this PR. To be clear, I meant the backend controller layer (the server URL endpoints that were added here), not the frontend controller -- does that clarify?

Reply:
But that's actually what I feel should be different - shouldn't it be a frontend check rather than a controller check?

Reply:
Also, if it should be in a backend controller layer, it should be in the question player controller not the review test controller.

Reply:
Oh wait, sorry, you're right. I was thinking about the backend controller that @sophiewu6 added in https://github.com/oppia/oppia/pull/6850/files#diff-9b80879a6a1dcf72afcadb8b5870fe46R77 but I just realized it doesn't actually call this function. So you're right @vinitamurthi, sorry! 

Could we please add a task to the question player team to make sure this check gets added before releasing?

Also it should be both a frontend check and a backend controller check. Technically it's possible to ping the server without going through the frontend, so we need to harden the backend against arbitrary input.

Finally @sophiewu6 I think just the third paragraph of my original comment applies here then: 

> Then, at this level, do a similar check but use an if statement and raise Exception if the requested count is above the limit, since nothing should ever get through to this point in normal system operation.

Reply:
Created an issue #6928 for the frontend/backend controllers. @sophiewu6 can you add the exception Sean mentioned in this PR?

Reply:
Done. Also added a test for that.

I had to push package-lock.json again, but once the review is done, I will revert it back.